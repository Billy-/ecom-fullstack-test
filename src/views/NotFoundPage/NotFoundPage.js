import React from 'react';

import styles from './NotFoundPage.scss';

const NotFoundPage = ({ location: { pathname } }) => (
  <div className={styles.container}>
    <h2>Sorry, the page at "{pathname}" was not found</h2>
  </div>
);

export default NotFoundPage;
