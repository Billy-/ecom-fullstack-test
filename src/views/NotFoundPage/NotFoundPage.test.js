import React from 'react';
import renderer from 'react-test-renderer';

import NotFoundPage from './NotFoundPage';

const props = {
    location: {
        pathname: 'foo'
    }
};

describe('NotFoundPage', () => {
    it('renders without throwing', () => {
        renderer.create(<NotFoundPage {...props} />);
    });

    it('displays the pathname from react-router somewhere in the error message', () => {
        const instance = renderer.create(<NotFoundPage {...props} />);
        const message = instance.root.findByType('h2').props.children;
        expect(message.includes(props.location.pathname)).toBe(true);
    });

    it('matches snapshot', () => {
        const tree = renderer
            .create(<NotFoundPage {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
