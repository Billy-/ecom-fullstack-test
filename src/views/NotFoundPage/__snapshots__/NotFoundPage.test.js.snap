// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage matches snapshot 1`] = `
<div
  className="container"
>
  <h2>
    Sorry, the page at "
    foo
    " was not found
  </h2>
</div>
`;
